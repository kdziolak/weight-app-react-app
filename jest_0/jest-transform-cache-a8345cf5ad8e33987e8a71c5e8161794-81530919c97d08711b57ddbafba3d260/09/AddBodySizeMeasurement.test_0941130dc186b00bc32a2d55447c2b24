'use strict';var _jsxFileName = '/home/kamil/Documents/Projects/weight-app-react-app/src/tests/components/AddBodySizeMeasurement.test.js';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _AddBodySizeMeasurement = require('../../components/container/AddBodySizeMeasurement/AddBodySizeMeasurement');var _AddBodySizeMeasurement2 = _interopRequireDefault(_AddBodySizeMeasurement);
var _enzyme = require('enzyme');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('AddBodySizeMeasurement component', () => {
    it('render component', () => {
        const wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_AddBodySizeMeasurement2.default, { __source: { fileName: _jsxFileName, lineNumber: 7 }, __self: undefined }));
    });
});