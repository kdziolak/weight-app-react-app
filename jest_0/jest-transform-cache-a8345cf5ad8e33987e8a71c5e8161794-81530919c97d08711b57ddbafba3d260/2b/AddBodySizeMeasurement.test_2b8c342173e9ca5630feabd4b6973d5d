'use strict';var _jsxFileName = '/home/kamil/Documents/Projects/weight-app-react-app/src/__tests__/AddBodySizeMeasurement.test.js';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _reactTestingLibrary = require('react-testing-library');
var _AddBodySizeMeasurement = require('../components/container/AddBodySizeMeasurement/AddBodySizeMeasurement');var _AddBodySizeMeasurement2 = _interopRequireDefault(_AddBodySizeMeasurement);
var _HeaderTitle = require('../components/presentational/HeaderTitle/HeaderTitle');var _HeaderTitle2 = _interopRequireDefault(_HeaderTitle);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

afterEach(_reactTestingLibrary.cleanup);

test('first element is a div', () => {var _render =
    (0, _reactTestingLibrary.render)(_react2.default.createElement(_AddBodySizeMeasurement2.default, { __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: undefined }));const container = _render.container;
    expect(container.firstChild.nodeName).toBe('DIV');
});

test('render <HeaderTitle/> component', () => {var _render2 =
    (0, _reactTestingLibrary.render)(_react2.default.createElement(_AddBodySizeMeasurement2.default, { __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: undefined }));const container = _render2.container;
    const div = container.incl;
    expect(div.nodeName).toBe('DIV');
});