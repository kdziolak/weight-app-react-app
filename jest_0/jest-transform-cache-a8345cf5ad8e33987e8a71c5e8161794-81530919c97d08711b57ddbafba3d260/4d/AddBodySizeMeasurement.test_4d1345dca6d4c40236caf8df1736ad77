'use strict';var _jsxFileName = '/home/kamil/Documents/Projects/weight-app-react-app/src/__tests__/AddBodySizeMeasurement.test.js';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _enzyme = require('enzyme');
var _AddBodySizeMeasurement = require('../components/container/AddBodySizeMeasurement/AddBodySizeMeasurement');var _AddBodySizeMeasurement2 = _interopRequireDefault(_AddBodySizeMeasurement);
var _HeaderTitle = require('../components/presentational/HeaderTitle/HeaderTitle');var _HeaderTitle2 = _interopRequireDefault(_HeaderTitle);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}


describe('<App />', () => {
    let appWrapper;
    let appInstance;
    const app = (disableLifecycleMethods = false) =>
    (0, _enzyme.shallow)(_react2.default.createElement(_AddBodySizeMeasurement2.default, { __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: undefined }), { disableLifecycleMethods });

    beforeEach(() => {
        appWrapper = app();
        appInstance = appWrapper.instance();
    });

    afterEach(() => {
        appWrapper = undefined;
        appInstance = undefined;
    });

    it('renders without crashing', () => {
        expect(app().exists()).toBe(true);
    });

    it('renders a div', () => {
        expect(
        appWrapper.
        first().
        type()).
        toBe('div');
    });

    describe('the rendered div', () => {
        const div = () => appWrapper.first();

        it('contains everything else that gets rendered', () => {
            // Podczas wywoływania children() na wrapperze, enzyme pomija w wyszukiwaniu zewnętrzny węzeł. 
            // Dzięki temu możemy sprawdzić czy zawartość diva
            // faktycznie zawiera wszystko co renderuje <App />
            expect(div().children()).toEqual(appWrapper.children());
        });
    });
    it('renders <HeaderTitle />', () => {
        // Metoda find zwraca listę odnalezionych komponentów.
        // Najłatwiej sprawdzić czy komponent został wyrenderowany 
        // poprzez prostą asercję jej długości.
        expect(appWrapper.find(_HeaderTitle2.default).length).toBe(1);
    });
    describe('the rendered <Header />', () => {
        const header = () => appWrapper.find(_HeaderTitle2.default);
        // headerNumber={3} content='Add new weight' classes='blue-text text-darken-1' 

        it('"headerNumber" prop has to be equle 3 and resives', () => {
            expect(header().prop('headerNumber')).toBe(3);
        });
        it('"content" prop cannot to be empty string', () => {
            expect(header().prop('content')).not.toBe('');
        });
        it('"classes" prop has to h', () => {
            expect(header().prop('content')).not.toBe('');
        });
    });

});