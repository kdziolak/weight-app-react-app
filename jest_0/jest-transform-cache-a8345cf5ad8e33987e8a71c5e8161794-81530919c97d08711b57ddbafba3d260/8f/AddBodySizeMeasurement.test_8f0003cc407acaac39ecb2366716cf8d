'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _reactTestingLibrary = require('react-testing-library');
var _AddBodySizeMeasurement = require('../components/container/AddBodySizeMeasurement/AddBodySizeMeasurement');var _AddBodySizeMeasurement2 = _interopRequireDefault(_AddBodySizeMeasurement);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

afterEach(_reactTestingLibrary.cleanup);

test('without crashing', () => {
    expect(true).toBeTruthy();
});