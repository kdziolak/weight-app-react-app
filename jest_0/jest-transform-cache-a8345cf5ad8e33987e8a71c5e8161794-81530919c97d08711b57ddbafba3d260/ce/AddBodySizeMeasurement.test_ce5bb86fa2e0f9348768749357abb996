'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _reactTestingLibrary = require('react-testing-library');
var _ = require('.');var _2 = _interopRequireDefault(_);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}