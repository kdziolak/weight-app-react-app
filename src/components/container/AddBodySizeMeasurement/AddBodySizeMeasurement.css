.add-body-size-img-scale {
    width: 100%;
    display: flex;
    justify-content: center;
}

.add-body-size-img-scale img{
    display: block;
    height: 70vh;
}