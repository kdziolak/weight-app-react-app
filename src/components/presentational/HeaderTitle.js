import React from 'react'

export default function HeaderTitle(props) {
    switch(props.headerNumber){
        case 1: 
            return <h1 className={props.classes}>{props.content}</h1>
        case 2: 
            return <h2 className={props.classes}>{props.content}</h2>
        case 3: 
            return <h3 className={props.classes}>{props.content}</h3>
        case 4: 
            return <h4 className={props.classes}>{props.content}</h4>
        case 5: 
            return <h5 className={props.classes}>{props.content}</h5>
        case 6: 
            return <h6 className={props.classes}>{props.content}</h6>
    }
    return <h1 className={props.classes}>{props.content}</h1>;
}
