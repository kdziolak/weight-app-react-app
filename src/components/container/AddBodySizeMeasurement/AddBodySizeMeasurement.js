import React, { Component } from 'react';
import HeaderTitle from '../../presentational/HeaderTitle/HeaderTitle';

class AddBodySizeMeasurement extends Component {
    render() {
        return (
            <div data-testid="add-body-size-measurement-text" > <HeaderTitle headerNumber={3} content='Add new weight' classes='blue-text text-darken-1' /></div >
        );
    }

}

export default AddBodySizeMeasurement;