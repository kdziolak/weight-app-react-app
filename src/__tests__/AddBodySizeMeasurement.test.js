import React from 'react';
import { shallow, mount, render } from 'enzyme';
import AddBodySizeMeasurement from '../components/container/AddBodySizeMeasurement/AddBodySizeMeasurement';
import HeaderTitle from '../components/presentational/HeaderTitle/HeaderTitle';


describe('<App />', () => {
    let appWrapper;
    let appInstance;
    const app = (disableLifecycleMethods = false) =>
        shallow(<AddBodySizeMeasurement />, { disableLifecycleMethods });

    beforeEach(() => {
        appWrapper = app();
        appInstance = appWrapper.instance();
    });

    afterEach(() => {
        appWrapper = undefined;
        appInstance = undefined;
    });

    it('renders without crashing', () => {
        expect(app().exists()).toBe(true);
    });

    it('renders a div', () => {
        expect(
            appWrapper
                .first()
                .type(),
        ).toBe('div');
    });

    describe('the rendered div', () => {
        const div = () => appWrapper.first();

        it('contains everything else that gets rendered', () => {
            // Podczas wywoływania children() na wrapperze, enzyme pomija w wyszukiwaniu zewnętrzny węzeł. 
            // Dzięki temu możemy sprawdzić czy zawartość diva
            // faktycznie zawiera wszystko co renderuje <App />
            expect(div().children()).toEqual(appWrapper.children());
        });
    });
    it('renders <HeaderTitle />', () => {
        // Metoda find zwraca listę odnalezionych komponentów.
        // Najłatwiej sprawdzić czy komponent został wyrenderowany 
        // poprzez prostą asercję jej długości.
        expect(appWrapper.find(HeaderTitle).length).toBe(1);
    });
    describe('the rendered <HeaderTitle />', () => {
        const header = () => appWrapper.find(HeaderTitle);
        // headerNumber={3} content='Add new weight' classes='blue-text text-darken-1' 

        it('"headerNumber" prop has to be equle 3 and resives', () => {
            expect(header().prop('headerNumber')).toBe(3)
        });
        it('"content" prop cannot to be empty string', () => {
            expect(header().prop('content')).not.toBe('')
        });
        it('"classes" prop has to has a "blue-text text-darken-1" as a string.', () => {
            expect(header().prop('classes')).toBe('blue-text text-darken-1')
        });
    });

});

