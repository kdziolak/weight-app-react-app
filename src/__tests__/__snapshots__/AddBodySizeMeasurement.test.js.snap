// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <HeaderTitle/> component 1`] = `
HTMLCollection {
  "0": <h1 />,
  Symbol(HTMLCollection internal slots): Object {
    "conflictElements": Object {},
    "element": <div
      data-testid="add-body-size-measurement-text"
>
      <h1 />
</div>,
    "keys": Array [],
    "length": 1,
    "query": [Function],
    "version": 4,
  },
}
`;
